core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/person/child/EditChildProfileViewModel.kt

```diff
@@ -0,0 +1,177 @@
+import com.ustadmobile.core.impl.locale.entityconstants.PersonConstants
+import com.ustadmobile.core.impl.nav.UstadSavedStateHandle
+import com.ustadmobile.core.util.MessageIdOption2
+import com.ustadmobile.core.util.ext.onActiveEndpoint
+import com.ustadmobile.core.viewmodel.UstadEditViewModel
+import com.ustadmobile.door.ext.doorPrimaryKeyManager
+import com.ustadmobile.lib.db.entities.Person
+import com.ustadmobile.lib.db.entities.Person.Companion.GENDER_UNSET
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import org.kodein.di.DI
+import org.kodein.di.instance
+
+data class EditChildProfileUiState(
+    val person: Person? = null,
+
+    val dateOfBirthError: String? = null,
+
+    val genderOptions: List<MessageIdOption2> = PersonConstants.GENDER_MESSAGE_IDS_AND_UNSET,
+
+    val genderError: String? = null,
+
+    val firstNameError: String? = null,
+
+    val lastNameError: String? = null,
+
+    )
+
+class EditChildProfileViewModel(
+    di: DI,
+    savedStateHandle: UstadSavedStateHandle,
+) : UstadEditViewModel(di, savedStateHandle, DEST_NAME) {
+
+    private val _uiState = MutableStateFlow(
+        EditChildProfileUiState()
+    )
+    private val addNewPersonUseCase: AddNewPersonUseCase by di.onActiveEndpoint().instance()
+
+    val uiState: Flow<EditChildProfileUiState> = _uiState.asStateFlow()
+
+    private val genderConfig: GenderConfig by instance()
+
+    private fun EditChildProfileUiState.hasErrors(): Boolean {
+        return dateOfBirthError != null ||
+                firstNameError != null ||
+                lastNameError != null ||
+                genderError != null
+    }
+
+    init {
+        _appUiState.update { prev ->
+            prev.copy(
+
+                title = systemImpl.getString(MR.strings.child_profile),
+                hideBottomNavigation = true,
+            )
+        }
+        _uiState.update { prev ->
+            prev.copy(
+                person = Person(),
+                genderOptions = genderConfig.genderMessageIdsAndUnset,
+
+                )
+        }
+
+        _appUiState.update { prev ->
+            prev.copy(
+                actionBarButtonState = ActionBarButtonUiState(
+                    visible = true,
+                    text = systemImpl.getString(MR.strings.done),
+                    onClick = this@EditChildProfileViewModel::onClickDone
+                )
+            )
+        }
+        viewModelScope.launch {
+            val selectedContentEntry = savedStateHandle.getJson(
+                ARG_SELECTED_PERSON_ENTRY, Person.serializer()
+            )
+
+            val loadedEntity = loadEntity(
+                serializer = Person.serializer(),
+                makeDefault = {
+                    val newUid = activeDb.doorPrimaryKeyManager.nextIdAsync(Person.TABLE_ID)
+                    Person(
```

Comment:
This should only create a blank new person. If a previous value was passed in an argument, then makeDefault is not called.